src/main/java/org/zalando/intellij/swagger/file/FileDetector.java

```diff
@@ -11,34 +12,46 @@
 public class FileDetector {
 
     private static final String SWAGGER_KEY = "swagger";
+    private static final String SWAGGER_VERSION = "2";
     private static final String OPEN_API_KEY = "openapi";
+    private static final String OPEN_API_VERSION = "3";
 
     public boolean isMainSwaggerJsonFile(final PsiFile psiFile) {
-        return hasJsonRootKey(psiFile, String.format("$.%s", SWAGGER_KEY));
+        return hasJsonRootKey(psiFile, String.format("$.%s", SWAGGER_KEY), SWAGGER_VERSION);
     }
 
     public boolean isMainSwaggerYamlFile(final PsiFile psiFile) {
-        return hasYamlRootKey(psiFile, String.format("$.%s", SWAGGER_KEY));
+        return hasYamlRootKey(psiFile, String.format("$.%s", SWAGGER_KEY), SWAGGER_VERSION);
     }
 
     public boolean isMainOpenApiJsonFile(final PsiFile psiFile) {
-        return hasJsonRootKey(psiFile, String.format("$.%s", OPEN_API_KEY));
+        return hasJsonRootKey(psiFile, String.format("$.%s", OPEN_API_KEY), OPEN_API_VERSION);
     }
 
     public boolean isMainOpenApiYamlFile(final PsiFile psiFile) {
-        return hasYamlRootKey(psiFile, String.format("$.%s", OPEN_API_KEY));
+        return hasYamlRootKey(psiFile, String.format("$.%s", OPEN_API_KEY), OPEN_API_VERSION);
     }
 
-    private boolean hasYamlRootKey(final PsiFile psiFile, final String lookupKey) {
+    private boolean hasYamlRootKey(final PsiFile psiFile, final String lookupKey, final String lookupVersion) {
         final Language language = psiFile.getLanguage();
-
-        return YAMLLanguage.INSTANCE.is(language) && new PathFinder().findByPathFrom(lookupKey, psiFile).isPresent();
+        if(!YAMLLanguage.INSTANCE.is(language)){
+            return false;
+        }
+        return new PathFinder().findByPathFrom(lookupKey, psiFile)
+                .filter(psiElement -> hasVersion(psiElement, lookupVersion)).isPresent();
     }
 
-    private boolean hasJsonRootKey(final PsiFile psiFile, final String lookupKey) {
+    private boolean hasJsonRootKey(final PsiFile psiFile, final String lookupKey, final String lookupVersion) {
         final Language language = psiFile.getLanguage();
+        if(!JsonLanguage.INSTANCE.is(language)){
+            return false;
+        }
+        return new PathFinder().findByPathFrom(lookupKey, psiFile)
+                .filter(psiElement -> hasVersion(psiElement, lookupVersion)).isPresent();
+    }
 
-        return JsonLanguage.INSTANCE.is(language) && new PathFinder().findByPathFrom(lookupKey, psiFile).isPresent();
+    private boolean hasVersion(final PsiElement psiElement, final String lookupVersion) {
+        return psiElement.getLastChild().getText().contains(lookupVersion);
```

Comment:
We should use optionals here, `getLastChild()` and `getText()` can return null values.

Reply:
We could also check that the value is an exact match, according to spec the value must be `2.0`.